pipeline/util/urllib.py

```diff
@@ -36,39 +38,44 @@ def download(
     Note: timestamps of remote files are not retained.
     """
 
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        tmpfilename = os.path.join(tmpdirname, filename)
-        finalfilename = os.path.join(destination_path, filename)
-
-        if not os.path.exists(destination_path):
-            os.makedirs(destination_path)
-
-        headers = {'User-Agent': user_agent}
-        if not force and os.path.exists(finalfilename):
-            pre_existing_file_last_modified = datetime.utcfromtimestamp(os.path.getmtime(finalfilename))
-            headers['If-Modified-Since'] = pre_existing_file_last_modified.strftime('%a, %d %b %Y %H:%M:%S GMT')
-
-        response = get(source, timeout=timeout, headers=headers, stream=True)
-        if response.status_code == 304:
-            # File not modified since last download
-            return
-
-        with open(tmpfilename, 'wb') as file:
-            for chunk in response.iter_content(chunk_size=1024):
-                if chunk:
-                    file.write(chunk)
-
-        last_modified = None
-        last_modified_str = response.headers.get('Last-Modified')
-        if last_modified_str:
-            last_modified = parsedate_to_datetime(last_modified_str)
-            os.utime(tmpfilename, (last_modified.timestamp(), last_modified.timestamp()))
-
-        if gzip:
-            with open(tmpfilename, 'rb') as f_in, gzip.open(tmpfilename + '.gz', 'wb') as f_out:
-                shutil.copyfileobj(f_in, f_out)
-                if last_modified:
-                    os.utime(tmpfilename + '.gz', (last_modified.timestamp(), last_modified.timestamp()))
-                os.replace(tmpfilename + '.gz', finalfilename + '.gz')
-
-        os.replace(tmpfilename, finalfilename)
+    tmpfilename = os.path.join(destination_path, filename + '.tmp')
```

Comment:
Using pathlib would be the cleaner way to handle file paths.

Reply:
Could you elaborate why you think it would be much cleaner and worth introducing another external dependency?